bgpd/bgp_zebra.h

```diff
@@ -23,6 +23,13 @@
 
 #include "vxlan.h"
 
+/* Macro to update bgp_original based on bpg_path_info */
+#define BGP_ORIGINAL_UPDATE(_bgp_orig, _mpinfo, _bgp)                          \
+	((_mpinfo->extra && _mpinfo->extra->bgp_orig                           \
```

Comment:
can you add a protection with _mpinfo.
unless I misread, mpinfo can be NULL in evpn case. no?

Reply:
In case of evpn, what we observed is _mpinfo->extra->bgp_orig can be NULL.
Earlier we were only checking for _mpinfo->extra, where as in the case of evpn,
we had to add NULL check for original bgp path info.

EVPN routes marked as imported routes, having bgp path info's extra,
where as they are not truly imported routes, so original bgp info will be null (.
While comparing nexthop's vrf_id, we were dereferencing null.

Adding to that within bgp_zebra_announce() where we are using this macro,
we are already checking for NULL check with for loop.
`for (mpinfo = info; mpinfo; mpinfo = bgp_path_info_mpath_next(mpinfo)) {`
--------
Having said that, we can always add NULL check,
which is always better thing to do.

Please let us know.